firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteOverlayMigrationManager.java

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import static com.google.firebase.firestore.util.Assert.fail;
+
+import androidx.annotation.VisibleForTesting;
+import com.google.firebase.firestore.auth.User;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.model.mutation.MutationBatch;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/** Manages SQLite data migration required by SDK version upgrades. */
+public class SQLiteOverlayMigrationManager implements OverlayMigrationManager {
+  private final SQLitePersistence db;
+
+  /**
+   * Creates a new data migration manager.
+   *
+   * @param persistence The underlying SQLite Persistence to use for data migrations.
+   */
+  public SQLiteOverlayMigrationManager(SQLitePersistence persistence) {
+    this.db = persistence;
+  }
+
+  @Override
+  public void run() {
+    if (Persistence.OVERLAY_SUPPORT_ENABLED == false || !hasPendingOverlayMigration()) {
```

Comment:
`pendingOverlayMigrations` should not be run if overlay disabled. Because it might be trying to access a table that does not exist.

This goes away in next PR anyways.